File: Python/Programmers/Leve1/정수 제곱근 판별.py
# def solution(n):
#     return (n**0.5 + 1)**2 if n**0.5 == int(n**0.5) else -1 # 정수 x가 n의 제곱근인지를 확인하려면 n**0.5와 int(n**0.5)가 같은지 확인하면 됨

def solution(n):
    return (n**0.5+1)**2 if n**0.5 % 1 ==0 else -1 # n**0.5 % 1로 제곱근이 정수인지 아닌지 판단 가능

File: JavaScript/Programmers/Level 1/정수 제곱근 판별.js
// n이 x의 제곱인지 아닌지 확인
// parseInt 함수를 사용하여 n의 제곱근이 정수인지 확인

// function solution(n) {
//     // n의 제곱근이 정수면 n의 제곱근 + 1의 제곱 return, 아니면 -1 return
//     return parseInt(n ** 0.5) === n ** 0.5 ? (n ** 0.5 + 1) ** 2 : -1;
// }

function solution(n) {
    // % 연산자를 활용하여 1로 나눈 나머지를 확인하면 n의 제곱근이 정수인지 확인할 수 있음
    return !((n ** 0.5) % 1) ? (n ** 0.5 + 1) ** 2 : -1;
}
